src/components/com_tjdashboard/administrator/tables/widgets.php

```diff
@@ -55,4 +56,38 @@ public function check()
 
 		return parent::check();
 	}
+
+	/**
+	 * Method to delete a Widget.
+	 *
+	 * @param   int  $pk  Primary key value to delete. Optional
+	 * 
+	 * @return  boolean  True on success.
+	 *
+	 * @since   1.0.0
+	 */
+	public function delete($pk = null)
+	{
+		try
+		{
+			$widget = TjdashboardWidget::getInstance($pk);
+
+			if ($widget->core != 1)
+			{
+				$result = parent::delete($pk);
+			}
+			elseif ($widget->core == 1)
+			{
+				$this->setError(JText::_('COM_TJDASHBOARD_DEFAULT_WIDGETS_DELETE_ERROR_MESSAGE'));
+			}
+
+			return $result;
+		}
+		catch (Exception $e)
+		{
+			$this->setError(JText::_('COM_TJDASHBOARD_WIDGETS_DELETE_ERROR_MESSAGE'));
```

Comment:
What is our approach on catching exceptions ? 

Shouldn't we know what kind of exceptions may be thrown and set an appropriate error. It's fine to set a default error also, but can't we catch other Exceptions first ?

Reply:
Agree. @shitalkakade24 catch database exceptions explicit.

Reply:
If we remove try catch &t in case any exception is occure then why we show sql error exception to user